Sources/NIOConcurrencyHelpers/NIOAtomic.swift

```diff
@@ -129,6 +129,29 @@ extension UInt64: NIOAtomicPrimitive {
     public static let nio_atomic_store                        = catmc_nio_atomic_unsigned_long_long_store
 }
 
+#if os(Windows)
+extension Int: NIOAtomicPrimitive {
+    public typealias AtomicWrapper = catmc_nio_atomic_intptr_t
+    public static let nio_atomic_create_with_existing_storage = catmc_nio_atomic_intptr_t_create_with_existing_storage
+    public static let nio_atomic_compare_and_exchange         = catmc_nio_atomic_intptr_t_compare_and_exchange
+    public static let nio_atomic_add                          = catmc_nio_atomic_intptr_t_add
+    public static let nio_atomic_sub                          = catmc_nio_atomic_intptr_t_sub
+    public static let nio_atomic_exchange                     = catmc_nio_atomic_intptr_t_exchange
+    public static let nio_atomic_load                         = catmc_nio_atomic_intptr_t_load
+    public static let nio_atomic_store                        = catmc_nio_atomic_intptr_t_store
+}
+
+extension UInt: NIOAtomicPrimitive {
+    public typealias AtomicWrapper = catmc_nio_atomic_uintptr_t
+    public static let nio_atomic_create_with_existing_storage = catmc_nio_atomic_uintptr_t_create_with_existing_storage
```

Comment:
@compnerd can’t we use the intptr_t version for all platforms?

Reply:
Alas, if we could.  Although `intptr_t` would be *better* as it works everywhere and is ABI compatible at C, the compiler remaps it differently and does not perform a type conversion between the two, so I don't think that we can.

Reply:
If you don't mind, I think I would. :smile:

Reply:
@compnerd what do you mean with "remaps it differently"?

Reply:
@weissi there are a number of builtin type mappings.  `UInt` is mapped to `long` on LP64 and `intptr_t` on LLP64.  There are no conversions that occur between the types like in C, so it would fail to construct the type.

Reply:
@compnerd ah, understood. Could we then maybe simplify the code and put something like

```
#if os(Windows)
typealias UIntMapping = catmc_nio_atomic_uintptr_t
#else
typealias UIntMapping = catmc_nio_atomic_unsigned_long
#endif

and then use `UIntMapping` in the extensions?

Reply:
@weissi afraid not.  The `let`s which follow actually reference the type as well.  The only thing being duplicated here is: `extension UInt: NIOAtomicPrimitive {}` and `extension Int: NIOAtomicPrimitive {}`.

Reply:
I see, okay, I'm fine with this then. Thank you!